### PR TITLE
Map report for the report builder

### PR DESCRIPTION
**Pull request by snopoke**
@biyeun 
cc @NoahCarnahan @nickpell 
# homeweekhackathon

:fish:

### CODE REVIEW

File: corehq/apps/reports/urls.py [hunk 1]
```diff
@@ -33,11 +34,12 @@
     CustomConfigurableReportDispatcher.url_pattern(),
 
     url(r'^builder/select_type/$', ReportBuilderTypeSelect.as_view(), name=ReportBuilderTypeSelect.urlname),
-    url(r'^builder/(?P<report_type>list|chart|table|worker)/select_source/$', ReportBuilderDataSourceSelect.as_view(), name='report_builder_select_source'),
+    url(r'^builder/(?P<report_type>list|chart|table|worker|map)/select_source/$', ReportBuilderDataSourceSelect.as_view(), name='report_builder_select_source'),
```
Comment (dimagimon) [inline corehq/apps/reports/urls.py]: E501 line too long (160 > 115 characters)

File: corehq/apps/reports_core/static/reports_core/js/maps.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    fn.render = function (config, data, mapContainer) {
+        mapContainer.show();
+        mapContainer.empty();
+        fn.init_map(config, mapContainer[0]);
+        fn.initPopupTempate(config);
+
+
+        var points = _.compact(_.map(data, function(row){
+            var val = row[config.location_column];
+            if (val != null) {
```
Comment (dimagimon) [inline corehq/apps/reports_core/static/reports_core/js/maps.js]: Use '!==' to compare with 'null'.

**Comment by czue** [comment]
looking good! no blockers on my side

**Comment by biyeun** [comment]
:+1: lg2m

**Comment by czue** [comment]
thanks! will merge when tests pass

**Comment by snopoke** [comment]
updated to more the mapbox access token to settings

**Comment by czue** [comment]
:+1:

**Comment by snopoke** [comment]
:white_check_mark:

**Comment by NoahCarnahan** [comment]
:+1:

**Comment by benrudolph** [comment]
this is :candy:!

**Pull request merged**